examples/vrf_on_shared_vc/main.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+
+	metal "github.com/equinix-labs/metal-go/metal/v1"
+	"sigs.k8s.io/yaml"
+)
+
+func main() {
+	var projectId string
+	flag.StringVar(&projectId, "projectId", "", "")
+	flag.Parse()
+
+	configuration := metal.NewConfiguration()
+	configuration.Debug = true
+	configuration.AddDefaultHeader("X-Auth-Token", os.Getenv("METAL_AUTH_TOKEN"))
+	api_client := metal.NewAPIClient(configuration)
+	ctx := context.Background()
+	metro := "da"
+
+	createProjectRequest := metal.ProjectCreateFromRootInput{
+		Name: "ctreatman-test-vrf",
+	}
+
+	project, r, err := api_client.ProjectsApi.CreateProject(ctx).ProjectCreateFromRootInput(createProjectRequest).Execute()
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error when calling `ProjectsApi.CreateProject`: %v\n", err)
+		fmt.Fprintf(os.Stderr, "Full HTTP response: %v\n", r)
+		panic(err)
+	}
+
+	createVrfRequest := metal.VrfCreateInput{
+		Name:     "my-test-vrf-shared",
+		Metro:    metro,
+		IpRanges: []string{"10.0.0.0/16"},
+	}
+
+	vrf, r, err := api_client.VRFsApi.CreateVrf(ctx, project.GetId()).VrfCreateInput(createVrfRequest).Execute()
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error when calling `VRFsApi.CreateVrf`: %v\n", err)
+		fmt.Fprintf(os.Stderr, "Full HTTP response: %v\n", r)
+		panic(err)
+	}
+
+	ipReservationRequest := metal.RequestIPReservationRequest{}
+	ipReservationRequest.VrfIpReservationCreateInput = &metal.VrfIpReservationCreateInput{
+		Type:    "vrf",
+		Cidr:    24,
+		Network: "10.0.0.0",
+		VrfId:   vrf.GetId(),
+	}
+
+	block, r, err := api_client.IPAddressesApi.RequestIPReservation(ctx, project.GetId()).RequestIPReservationRequest(ipReservationRequest).Execute()
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error when calling `IPAddressesApi.RequestIPReservation`: %v\n", err)
+		fmt.Fprintf(os.Stderr, "Full HTTP response: %v\n", r)
+		panic(err)
+	}
+
+	createVlanRequest := metal.VirtualNetworkCreateInput{
+		Metro: &metro,
+	}
+	vlan, r, err := api_client.VLANsApi.CreateVirtualNetwork(ctx, project.GetId()).VirtualNetworkCreateInput(createVlanRequest).Execute()
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error when calling `VLANsApi.CreateVirtualNetwork`: %v\n", err)
+		fmt.Fprintf(os.Stderr, "Full HTTP response: %v\n", r)
+		panic(err)
+	}
+
+	createGatewayRequest := metal.CreateMetalGatewayRequest{}
+	createGatewayRequest.VrfMetalGatewayCreateInput = &metal.VrfMetalGatewayCreateInput{
+		IpReservationId:  block.VrfIpReservation.GetId(),
+		VirtualNetworkId: vlan.GetId(),
+	}
+	createGatewayRequest.MetalGatewayCreateInput = &metal.MetalGatewayCreateInput{}
+
+	// TODO this returns a FindMetalGatewayById200Response; need to refactor to a named, reused component
```

Comment:
The spec section that corresponds to `FindMetalGatewayById200Response` is here: https://github.com/equinix-labs/metal-go/blob/main/spec/oas3.fetched/paths/metal-gateways/id.yaml#L56-L58

```yaml
oneOf:
- $ref: '../../components/schemas/MetalGateway.yaml'
- $ref: '../../components/schemas/VrfMetalGateway.yaml'
```

This chunk should be moved to a file under `components/schemas/`, but we'd need a good name for it.  Maybe rename the existing `MetalGateway.yaml` to `NonVrfMetalGateway.yaml`?  Is there a better name we could come up with?  Names like `GenericMetalGateway.yaml` (for the new component) have caused concern in the past, but I'm not sure how to give the existing `MetalGateway.yaml` a more specific name without referencing VRF.

Reply:
Moved to a separate issue; refactoring this probably doesn't make it any easier / more possible to create a VRF on shared VC example.